icefoganalytics/elcc-data-management
# ELCC-41: Update Reconciliation Tab To Include Employee Benefit Expenses

Fixes https://yg-hpw.atlassian.net/browse/ELCC-41

Relates to:

- [2022-23 Child Care Centres E-H](https://icefoganalyticscom.sharepoint.com/:x:/s/Project_Files/EXZzX-POQVxCiG_i7GLUBBIBGZEHHVaipJeyFUZCUx0ciA?e=hf8oZj&nav=MTJfRTk4X3swMDAwMDAwMC0wMDAxLTAwMDAtMDIwMC0wMDAwMDAwMDAwMDB9) -> Employee Benefits

# Context

Now that the Employee Benefits have been broken out into their own model, they are no longer reflected as expenses in the Child Care Centres → Summary → Reconciliation tab.

# Implementation

Simplify serializers implementation.
Use "Paid amount" as expense value in Child Care Centres → Summary → Reconciliation tab.
Rebuild CurrencyInput _again_ in hopes of having it behave as expected. Fixed bug where value was lost when tabbing over component, even if no changes where made.

# Screenshots

Estimated Employee Benefit Expense
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/7c5af9da-7296-4cf1-9389-ff6a64c8c474)
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/5362dc33-4734-4386-9391-4cb6cf9bb4fc)
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/ad46edcd-8443-4b27-baa6-4b05e95f3580)

Actual Employee Benefit Expense
> There is no tooltip when expense is from "actual", rather than estimated value.
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/cfc9ae90-24ce-404d-a592-ee31c5100cce)
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/ca724f48-f56d-40b3-a79a-3b052d55fcf9)

# Testing Instructions

1. Boot the app via `dev up`
2. Check that the app complies, and that you can log in at http://localhost:8080.
3. From the /dashboard page, click on the “Education is Currently Funding” card.
4. Click on a child care centre from the list to open the details panel on the right hand side of the page.
5. Click the “view details” button in the bottom right hand corner of the page.
6. From the detailed child care centre view, select the Employees tab. URL should look something like http://localhost:8080/child-care-centres/1/2023-24/employees/april
7. Enter values in all the Employee Benefits fields, but only in the "Est. Total" column, then click the "Save" button.
8. Next go to the "Summary" -> "Reconciliation" tab and check that the Expenses (April) column matches the "Paid Amount" value from the Employee Benefits section.
9. Check that there is an asterisk thing, that when hovered, show a note that this is based on an estimate value.
10. Go back to the "Employees" tab, and fill in the "Act. Total" column, then save.
11. Go back to the "Summary" -> "Reconciliation" tab, again, and check that the Expenses (April) column now matches the "Paid Amount" (actual) value, and no longer has an asterisk by it.
